component/wallet-cli/pkg/walletrunner/wallet_runner_oidc4ci.go

```diff
@@ -229,11 +232,192 @@ func (s *Service) RunOIDC4CI(config *OIDC4CIConfig, hooks *Hooks) error {
 	return nil
 }
 
+var matchRegex = regexp.MustCompile(oidc4ci.WalletInitFlowClaimRegex)
+
+func extractIssuerUrlFromScopes(scopes []string) (string, error) {
+	for _, scope := range scopes {
+		if matchRegex.MatchString(scope) {
```

Comment:
What's the idea behind this logic, where you only look for the first scope that matches? (Note that I'm not familiar with it's intended usage, which might be why I'm confused)

Reply:
As per requrement from @sudeshrshetty  one of the scopes will be URL to issuer, if we can find this URL - it means that we will start processing for wallet initiated flow